test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs

```diff
@@ -229,7 +234,34 @@ private bool CompareDatabaseSchemas(string databaseName1, string databaseName2)
                         (d.TargetObject?.ObjectType.Name == i.type && d.TargetObject?.Name?.ToString() == i.name)))
                 .ToList();
 
-            return remainingDifferences.Count == 0;
+            bool unexpectedDifference = false;
+            foreach (SchemaDifference schemaDifference in remainingDifferences)
+            {
+                if (schemaDifference.Name == "SqlTable" &&
+                    (schemaDifference.SourceObject.Name.ToString() == "[dbo].[DateTimeSearchParam]" ||
+                    schemaDifference.SourceObject.Name.ToString() == "[dbo].[StringSearchParam]"))
+                {
+                    foreach (SchemaDifference child in schemaDifference.Children)
+                    {
+                        if (child.TargetObject == null && child.SourceObject == null && (child.Name == "PartitionColumn" || child.Name == "PartitionScheme"))
+                        {
+                            // The ParitionColumn and the PartitionScheme come up in the differences list even though
+                            // when digging into the "difference" object the values being compared are equal.
+                            continue;
+                        }
+                        else
+                        {
+                            unexpectedDifference = true;
+                        }
+                    }
+                }
+                else
+                {
+                    unexpectedDifference = true;
+                }
+            }
+
+            return !unexpectedDifference;
```

Comment:
nit: I think we should move this logic into another function, so its clear that this is special. i.e. `unexpectedDifference = IgnorePartitionColumn(remainingDifferences);`

Reply:
What about adding a comment before the first foreach? I don't like the feel of having a method that is only used in one place for a non-reusable reason.